intake_esm/core.py

```diff
@@ -82,6 +82,66 @@ def __init__(self, esmcol_path, progressbar=True, log_level='CRITICAL', **kwargs
         self.cdf_kwargs = None
         self.preprocess = None
         self.aggregate = None
+        self.sep = sep
+
+    def __len__(self):
+        return len(self.df)
```

Comment:
Thinking about this a bit, I think this should be the number of datasets, not the number of rows in the catalog_file. Today, this would be essentially `len(self.to_dataset_dict())` -- but I have more comments on this point.